### PR TITLE
refactor: Reorganize Build Process

### PR DESCRIPTION
**Pull request by m2Giles**
This significantly reworks the build system for Bluefin/Aurora.

Changes:
Consolidates Workflows
Removes Surface and Asus images in favor of a single HWE image
Rewrite of local dev Justfile
Reorganization and Consolidation of build files

Add new files

Closes: #1841

### CODE REVIEW

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    
+permissions:
+    contents: read
+    packages: write
+    id-token: write
+
+jobs:
+    build-image-gts:
+        uses: ./.github/workflows/build-image-gts.yml
+        secrets: inherit
```
Comment (p5) [inline .github/workflows/build-images.yml:13]:
```suggestion
    build-image-gts:
        uses: ./.github/workflows/build-image-gts.yml
        secrets: inherit
        strategy:
          matrix:
            base_name: [bluefin, bluefin-dx, aurora, aurora-dx]
```

File: .github/workflows/build-image-gts.yml [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
   workflow_dispatch:
+  workflow_call: 
 
 jobs:
-  build:
-    name: build
+  build-image-gts:
+    name: Build GTS Images
     uses: ./.github/workflows/reusable-build.yml
     secrets: inherit
```
Comment (p5) [inline .github/workflows/build-image-gts.yml:18]:
```suggestion
  build-image-gts:
    name: Build GTS Images
    uses: ./.github/workflows/reusable-build.yml
    secrets: inherit
    strategy:
      matrix:
        image_flavor: [none, hwe, nvidia]
```

**Comment by m2Giles** [comment]
#1841
#1840

**Comment by m2Giles** [comment]
#1841
#1840

**Comment by castrojo** [comment]
@befanyt got any opinions here? If we're going to build a new world we might as well get new eyes. 😄

**Comment by m2Giles** [comment]
Changelogs script is in place. It needs to be wired up.

In testing I saw that GTS and Latest do not have dates associated with them. That is rectified in this PR.

**Comment by m2Giles** [comment]
Need to handle how stable-daily vs stable changelogs work.

Right now the workflow is called stable and we use that for changelogs. But we only want stable to get tagged weekly.

We could upload an artifact as a trigger. Or have some other way passing the fact that two changelogs are needed from resuable-build

**Comment by m2Giles** [comment]
Remaining:

Changelogs running for GTS and weekly stable only. If the workflow is GTS run thr changelog. If it's stable (not stable daily) make an output to tell to run changelogs.

Kernel Pinning. Have an input to reusable build for pinning a kernel.

**Comment by m2Giles** [comment]
I believe this is now ready for review.

**Review by bsherman — COMMENTED** [review]
Looking this over... this is a hugely valuable effort.

However, I think we should go one step further. We should be using the value of the just recipes in our github action workflow.

At the least, the `just build` recipe should be used in `reusable-build.yml` to build the images.
All the tagging logic probably makes sense to remain in the workflow since that's external to the container build itself. Similarly, the push operation doesn't need to be in just, and it ties closely to the tags.

We could possibly use `just build-rechunk` in the workflow as well... Looking for feedback from @m2Giles on that.

But the primary value from doing this is ability to build container images the same way both local and in github actions. That's huge!

Fixing a bug in the image build can be validated before pushing to a PR even.

**Comment by m2Giles** [comment]
I think the only thing needed to change to make that a reality is to make sure build container and rechunk can run completely as root.

The only major concern is not having enough space on builds when copying to the rootful podman store.

**Comment by bsherman** [comment]
> I think the only thing needed to change to make that a reality is to make sure build container and rechunk can run completely as root.

Ah, like an additional flag to the build recipe to run it as root? Yes, then it would match workflow behavior.

> The only major concern is not having enough space on builds when copying to the rootful podman store.

This should be addressed by running build as root, no?

**Comment by m2Giles** [comment]
Yes. I've already put in a bunch of sudoif's so this should be just testing to make sure it works as intended.

I may add the secureboot check to the justfile.

**Comment by bsherman** [comment]
After further review and discussion, while we DO want to make use of the new and improved just recipes within the github workflows, that effort can be iterated on. As is, this is already large and holding up some needed fixes.  So we should make an issue to reduce the workflow specific code by calling just instead.

I'm continuing to review the existing changes tonight. So far looking nice.

**Review by bsherman — DISMISSED** [review]
I have spent a bunch of time testing the justfile locally, looking through the build script changes (made a few changes there myself to fix numbering of build scripts), looking through the workflow yaml (fixed some naming conventions), and looking at the GREEN successful builds validating the podman build args.

I think this is looking good.


A great improvement and we'll keep on making it better.

Thanks @m2Giles !

**Review by bsherman — APPROVED** [review]

**Pull request merged**